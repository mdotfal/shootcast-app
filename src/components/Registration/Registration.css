* {
  box-sizing: border-box;
}

.reg-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

#email,
#password,
#password-repeat {
  width: 300px;
}

button {
  width: 100px;
}