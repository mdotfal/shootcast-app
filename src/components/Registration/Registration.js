import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Registration extends Component {

  state = {
    username: '',
    password: '',
    repeatPassword: '',
    isError: false,
    isSuccess: false
  }

  onUsernameChange = e => {
    this.setState({
      username: e.target.value
    })
  }

  onPasswordChange = e => {
    this.setState({
      password: e.target.value
    })
  }

  onRepeatPasswordChange = e => {
    this.setState({
      repeatPassword: e.target.value
    })
  }

  onSubmit = e => {
    e.preventDefault();
    if( this.state.password !== this.state.repeatPassword ) {
      return this.setState({
        isError: "Passwords do not match!"
      })
    };
    const result =  this.props.handleRegistration( this.state.username, this.state.password);
    if( !result ) {
      return this.setState({
        isError: 'Username already taken!'
      })
    } else {
      this.setState({
        isSuccess: true,
        isError: false
      })
    }
  }

  render() {
    return (
      <form onSubmit={ this.onSubmit }>
        <fieldset>
          <div className="reg-container">
            <h1>Register</h1>
            <p>Please fill in this form to create an account.</p>

            { this.state.isError && <div>{ this.state.isError }</div> }
        
            { this.state.isSuccess && <div>User created! <Link to='/' >Click here to login!</Link></div> }

            <label htmlFor="username"><b>Username</b></label>
            <input type="text" placeholder="Enter Username" name="username" id="username" onChange={ this.onUsernameChange } required />
        
            <label htmlFor="password"><b>Password</b></label>
            <input type="password" placeholder="Enter Password" name="password" id="password" onChange={ this.onPasswordChange } required />
        
            <label htmlFor="password-repeat"><b>Repeat Password</b></label>
            <input type="password" placeholder="Repeat Password" name="password-repeat" id="password-repeat" onChange={ this.onRepeatPasswordChange } required />
        
            
            <button type="submit" className="registerbtn">Register</button>
            <Link>
              <button type="button" className="cancelbtn">Cancel</button>
            </Link>
          </div>
        </fieldset>
      </form>
    );
  }
}

export default Registration;