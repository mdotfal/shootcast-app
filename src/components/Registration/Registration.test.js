/* eslint-disable jest/valid-describe */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import Registration from './Registration';

describe.only( 'Registration Component', () => {
  test( 'should render', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <Registration />
        </BrowserRouter>
      ).toJSON();
    expect( component ).toMatchSnapshot();
  })

  it( 'renders without crashing', () => {
    const div = document.createElement( 'div' );
    ReactDOM.render(
      <BrowserRouter>
        <Registration />
      </BrowserRouter>
      , div );
      ReactDOM.unmountComponentAtNode( div );
    })
})