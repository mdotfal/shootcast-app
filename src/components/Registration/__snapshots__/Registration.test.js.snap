// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Component should render 1`] = `
<form
  className="reg-form"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="reg-container"
    >
      <h1>
        Register
      </h1>
      <p>
        Please fill in this form to create an account.
      </p>
      <label
        htmlFor="username"
      >
        <b>
          Username
        </b>
      </label>
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Enter Username"
        required={true}
        type="text"
      />
      <br />
      <label
        htmlFor="password"
      >
        <b>
          Password
        </b>
      </label>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
      />
      <br />
      <label
        htmlFor="password-repeat"
      >
        <b>
          Repeat Password
        </b>
      </label>
      <input
        id="password-repeat"
        name="password-repeat"
        onChange={[Function]}
        placeholder="Repeat Password"
        required={true}
        type="password"
      />
      <br />
      <button
        className="registerbtn"
        type="submit"
      >
        Register
      </button>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="cancelbtn"
          type="button"
        >
          Cancel
        </button>
      </a>
    </div>
  </fieldset>
</form>
`;
